framework/external/easylogging/easylogging++.cc

```diff
@@ -1755,6 +1755,20 @@ unsigned long TypedConfigurations::getULong(std::string confVal) {
   return atol(confVal.c_str());
 }
 
+std::string TypedConfigurations::getDateTimeForFilename(const std::string &fmt) {
+  m_dateTimeMutex.lock();
+
+  if (!m_dateTimeCalculated) {
+    base::SubsecondPrecision ssPrec(3);
+    m_dateTimeNow = base::utils::DateTime::getDateTime(fmt.c_str(), &ssPrec);
+    m_dateTimeCalculated = true;
+  }
```

Comment:
You don't need 2 member variables here.
Checking if `m_dateTimeNow` is empty should be enough:
```suggestion
  if (m_dateTimeNow.empty()) {
    base::SubsecondPrecision ssPrec(3);
    m_dateTimeNow = base::utils::DateTime::getDateTime(fmt.c_str(), &ssPrec);
  }
```